internal/querycoordv2/observers/leader_observer.go

```diff
@@ -74,7 +74,10 @@ func (o *LeaderObserver) observeCollection(ctx context.Context, collection int64
 			dists := o.dist.SegmentDistManager.GetByShardWithReplica(ch, replica)
 			needLoaded, needRemoved := o.findNeedLoadedSegments(leaderView, dists),
 				o.findNeedRemovedSegments(leaderView, dists)
-			o.sync(ctx, leaderView, append(needLoaded, needRemoved...))
+			if len(needLoaded) > 0 || len(needRemoved) > 0 {
+				o.sync(ctx, leaderView, append(needLoaded, needRemoved...))
```

Comment:
`sync()` do nothing if the given slice is empty, this check seems to be redundant

Reply:
Sorry, I forgot to pull the latest branch. Thanks for the reminder and I'll close this pr.